### DISCUSSION TITLE
Drag and drop support for URLs

### DISCUSSION THREAD

**Discussion by hakonharnes** [ghostty-org/ghostty#2917] [Feature Requests, Ideas] [answered]
**Description**

Ghostty currently supports drag and drop for files (e.g., from Finder), but URLs dragged from browsers like Safari do not work as expected. When dragging a URL, the cursor changes to “+”, but nothing is pasted into the terminal.

Expanding drag-and-drop support for URLs would align Ghostty with common terminal behavior and unlock additional use cases. Here's an [overview of other terminal emulators' drag and drop capabilities](https://github.com/HakonHarnes/img-clip.nvim?tab=readme-ov-file#%EF%B8%8F-drag-and-drop).

**Steps to reproduce**

1.  Open Ghostty.
2. Drag a URL (e.g., an image link) from Safari into the terminal.
3.  Observe that the “+” cursor appears, but no action occurs.

**Expected behavior**

Dragged URLs should be pasted into the terminal, similar to how file paths are handled.

**Demonstration**

https://github.com/user-attachments/assets/3c889cab-78a2-4a8c-a3c9-3a8747377f5a

Apple Terminal (top) vs. Ghostty (bottom). Observe that the URL is pasted in Apple Terminal but not in Ghostty.

**Environment**

- MacOS Sequoia 15.1.1 
- Ghostty 0.1.0-main+ae2cf8dc 
- Safari 18.1.1 (20619.2.8.11.12)

**Comment by TheDevMinerTV** [comment]
I'd also love this for files drag'n dropped from my file manager. Konsole and Kitty also allow this.

**Comment by freb** [comment]
My main use case is dragging files out of the terminal, which doesn't seem to be supported. Kitty also doesn't support dragging files out and into a file browsers or web page, but it does allow you to change the default file handlers. So I have it open all files with dragon, which provides a draggable reference to the file.

**Comment by mitchellh** [comment, marked as answer]
This has worked for some time, not sure when it was resolved.
